top-k-frequent-elements/yolophg.js

```diff
@@ -0,0 +1,28 @@
+// Time Complexity: O(n)
+// Space Complexity: O(n)
+
+var topKFrequent = function(nums, k) {
+    const frequencyMap = new Map();
+    // for each number in the array, update frequency.
+    for (const num of nums) {
+        frequencyMap.set(num, (frequencyMap.get(num) || 0));
+    }
+
+    // create buckets where index represents frequency.
+    const buckets = Array(nums.length + 1).fill().map(() => []);
+    // place each number into the bucket corresponding to frequency.
+    for (const [num, frequency] of frequencyMap.entries()) {
+        buckets[frequency].push(num);
+    }
+
+    const result = [];
+    // iterate from the highest possible frequency down to the lowest.
+    for (let i = buckets.length - 1; i >= 0 && result.length < k; i--) {
+        if (buckets[i].length > 0) {
+            result.push(...buckets[i]);
+        }
```

Comment:
`if` 문이 굳이 필요할까 하는 생각이 들었어요. `result.push(...[])`는 아무 원소도 추가하지 않을테니까요.

```suggestion
        result.push(...buckets[i]);
```

Reply:
그러게요, 빈 배열인 경우에는 불필요한 연산을 피해가기 위한 예외처리였는데 다시 생각해보니 애초에 굳이 없어도 될 것 같습니다!
피드백 감사합니다~ 반영해서 적용하였습니다!